contracts/test/FactoryV2.cairo

```diff
@@ -0,0 +1,295 @@
+@external
+func initializer{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    pair_proxy_contract_class_hash: felt, pair_contract_class_hash: felt, fee_to_setter: felt
+) {
+
+    with_attr error_message("Factory::constructor::Fee Recipient Setter can not be zero") {
+        assert_not_zero(fee_to_setter);
+    }
+
+    with_attr error_message("Factory::constructor::Pair Proxy Contract Class Hash can not be zero") {
+        assert_not_zero(pair_proxy_contract_class_hash);
+    }
+    
+    with_attr error_message("Factory::constructor::Pair Contract Class Hash can not be zero") {
+        assert_not_zero(pair_contract_class_hash);
+    }
+
+    _fee_to_setter.write(fee_to_setter);
+    _pair_proxy_contract_class_hash.write(pair_proxy_contract_class_hash);
+    _pair_contract_class_hash.write(pair_contract_class_hash);
+    _num_of_pairs.write(0);
+    Proxy.initializer(fee_to_setter);
+    return ();
+}
+
+//
+// Getters
+//
+
+// @notice Get pair address for the pair of `token0` and `token1`
+// @param token0 Address of token0
+// @param token1 Address of token1
+// @return pair Address of the pair
+@view
+func get_pair{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    token0: felt, token1: felt
+) -> (pair: felt) {
+    return _pair.read(token0, token1);
+}
+
+// @notice Get all the pairs registered
+// @return all_pairs_len Length of `all_pairs` array
+// @return all_pairs Array of addresses of the registered pairs
+@view
+func get_all_pairs{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+    all_pairs_len: felt, all_pairs: felt*
+) {
+    alloc_locals;
+    let (num_pairs) = _num_of_pairs.read();
+    let (local all_pairs: felt*) = alloc();
+    let (all_pairs_end: felt*) = _build_all_pairs_array(0, num_pairs, all_pairs);
+    return (num_pairs, all_pairs);
+}
+
+// @notice Get the number of pairs
+// @return num_of_pairs
+@view
+func get_num_of_pairs{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+    num_of_pairs: felt
+) {
+    let (num_of_pairs) = _num_of_pairs.read();
+    return (num_of_pairs,);
+}
+
+// @notice Get fee recipient address
+// @return address
+@view
+func get_fee_to{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+    address: felt
+) {
+    return _fee_to.read();
+}
+
+// @notice Get the address allowed to change fee_to.
+// @return address
+@view
+func get_fee_to_setter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+    address: felt
+) {
+    return _fee_to_setter.read();
+}
+
+// @notice Get the class hash of the Pair contract which is deployed for each pair.
+// @return class_hash
+@view
+func get_pair_contract_class_hash{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    ) -> (class_hash: felt) {
+    return _pair_proxy_contract_class_hash.read();
+}
+
+@view
+func test_v2_contract{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
```

Comment:
This contract is exactly similar to `Factory.cairo`, except that it has this extra function. I don't understand the requirement of this contract. 
Also, why does it has the suffix "V2" in its name?